Sources/BraintreeCore/BTAPIClient.m

```diff
@@ -273,75 +274,55 @@ - (void)fetchPaymentMethodNonces:(BOOL)defaultFirst completion:(void (^)(NSArray
 #pragma mark - Remote Configuration
 
 - (void)fetchOrReturnRemoteConfiguration:(void (^)(BTConfiguration *, NSError *))completionBlock {
-    // Guarantee that multiple calls to this method will successfully obtain configuration exactly once.
+    // Fetches or returns the configuration and caches the response in the GET BTHTTP call if successful
     //
     // Rules:
     //   - If cachedConfiguration is present, return it without a request
-    //   - If cachedConfiguration is not present, fetch it and cache the succesful response
-    //     - If fetching fails, return error and the next queued will try to fetch again
-    //
-    // Note: Configuration queue is SERIAL. This helps ensure that each request for configuration
-    //       is processed independently. Thus, the check for cached configuration and the fetch is an
-    //       atomic operation with respect to other calls to this method.
-    //
-    // Note: Uses dispatch_semaphore to block the configuration queue when the configuration fetch
-    //       request is waiting to return. In this context, it is OK to block, as the configuration
-    //       queue is a background queue to guarantee atomic access to the remote configuration resource.
-    dispatch_async(self.configurationQueue, ^{
-        __block NSError *fetchError;
-
-        dispatch_semaphore_t semaphore = dispatch_semaphore_create(0);
```

Comment:
It's nice to see the semaphore code removed 🙌 . Was this the root cause?

Reply:
Having this is an semaphore was at least part of the root cause.